### PR TITLE
Add Dockerfile and initial params

### PR DESCRIPTION
**Pull request by GreatAlexander**

### CODE REVIEW

File: av_ouster_launch/launch/av_ouster.launch.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<launch>
+  <include file="$(find-pkg-share ros2_ouster)/launch/driver_launch.py">
+    <arg name="params_file" value="$(find-pkg-share av_ouster_launch)/params/av_ouster_config.yaml"/>
```
Comment (hect95) [inline av_ouster_launch/launch/av_ouster.launch.xml:3]: @GreatAlexander  I think this version does not run the nodes inside a container. Is up to us if we want to leave it like that for now. Anyways, we have some fmw updates to the ouster and other ROS2 repo testing

File: av_ouster_launch/package.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>av_ouster_launch</name>
+  <version>0.0.0</version>
+  <description>AV Ouster config and launch</description>
+  <maintainer email="alex.bordallo@ed.ac.uk">Alejandro Bordallo</maintainer>
```
Comment (hect95) [inline av_ouster_launch/package.xml:7]: Taking into account about the reviewer and author roles in the reviews, should I add myself as a maintainer or will you do it? :)

**Review by hect95 — CHANGES_REQUESTED** [review]

File: runtime.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+# Build docker image only up to base stage
+DOCKER_BUILDKIT=1 docker build \
+    -t av_ouster_launch_humble \
+    -f Dockerfile --target runtime .
+
+# Run docker image without volumes
+docker run -it --rm --net host \
+    -v /dev/shm:/dev/shm \
+    av_ouster_launch_humble $CMD
```
Comment (hect95) [inline runtime.sh:22]: As discussed docker image name should be: `av_ouster:latest`

File: dev.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    -t av_ouster_launch_humble \
+    -f Dockerfile --target dev .
+
+# Run docker image with local code volumes for development
+docker run -it --rm --net host \
+    -v /dev/shm:/dev/shm \
+    -v ./av_ouster_launch:/opt/ros_ws/src/av_ouster_launch \
+    -v ./ouster_msgs:/opt/ros_ws/src/ouster_msgs \
+    -v ./ros2_ouster:/opt/ros_ws/src/ros2_ouster \
+    av_ouster_launch_humble
```
Comment (hect95) [inline dev.sh:17]: As discussed docker image name should be: `av_ouster:latest`

**Review by hect95 — CHANGES_REQUESTED** [review]
More changes :)

File: dev.sh (hunk 4 above)
Comment (GreatAlexander) [reply dev.sh]: Done

File: runtime.sh (hunk 3 above)
Comment (GreatAlexander) [reply runtime.sh]: Done

File: av_ouster_launch/launch/av_ouster.launch.xml (hunk 1 above)
Comment (GreatAlexander) [reply av_ouster_launch/launch/av_ouster.launch.xml:3]: Indeed... Given it's 1 node to run as composable, it doesn't sound like it's worth the effort to rework the original driver to make it one.
Also this launch depends on a Python launchfile :vomiting_face: which also flags lots of issues with flake8 ros from pre-commit. Given the Python launchfile is imported from the driver repo, tempted to leave it as is and ignore in pre-commit, unless a brave soul wants to have a go at fixing it :sweat_smile:

Comment (hect95) [reply av_ouster_launch/launch/av_ouster.launch.xml:3]: Happy to leave it like that. I think the launch file  from the Ouster oficial repo is a bit cleaner. Anyway we will archive this repo soon anyways

**Review by hect95 — APPROVED** [review]
Good to go!

**Comment by GreatAlexander** [comment]
I'll merge this PR in the repo in case we need it, but we may well remove this fork in future.

**Pull request merged**